src/LSP/language-features/text-document.jl

```diff
@@ -0,0 +1,195 @@
+"""
```

Comment:
Following our rule for LSP organization, I think this file should live under `LSP/life-cycle-messages/`.

Reply:
Would it be better to add it to the `LSP/capabilities.jl` file? `TextDocumentClientCapabilities` are described right before `ClientCapabilities` in the LSP specification.

Reply:
yeah that sounds fine too